cgi-bin/LJ/Support.pm

```diff
@@ -907,6 +907,40 @@ sub touch_request
     return 1;
 }
 
+# Extra email addresses are stored as support properties
+# - nb_extra_addresses: number of extra addresses (if not present, 0)
+# - extra_address_$n: extra address $n (0<=$n<nb_extra_addresses)
+
+sub add_email_address {
+    my ( $sp, $address ) = @_;
+
+    # Already present?
+    return if grep { $_ eq $address } @{all_email_addresses( $sp )};
+
+    # Add
+    my $props = load_props( $sp->{spid} + 0 );
+    my $nb_extra_addresses = $props->{nb_extra_addresses} || 0;
+    set_prop( $sp->{spid}, 'nb_extra_addresses', $nb_extra_addresses + 1 );
```

Comment:
Clever!

Now I'm torn here because this does work, but it breaks a sort of hidden expectation of props having definite keys.

*But*, and importantly, from the point of view of *not* losing data this is better than concatenating extra email addresses into one property value.